src/utils/fileDialog.ts

```diff
@@ -0,0 +1,39 @@
+interface IFileDialogOptions {
+  onChange: (files?: FileList) => void;
+  multiple?: boolean;
+  accept?: string;
+}
+
+const defaultDialogOptions: IFileDialogOptions = {
+  onChange: () => {},
+  multiple: false,
+  accept: '*',
+};
+
+function createInputEl(): HTMLInputElement {
+  const input = document.createElement('input');
+  input.type = 'file';
+  input.hidden = true;
+  input.className = 'hidden';
+  document.body.appendChild(input);
+  return input;
+}
+
+export function openFileDialog(opts: IFileDialogOptions = defaultDialogOptions): Promise<void> {
+  return new Promise<void>((resolve) => {
+    const inputEl = createInputEl();
+    const options = { ...opts };
+
+    inputEl.accept = options.accept!;
+    inputEl.multiple = options.multiple!;
```

Comment:
If you will follow my suggestions the excalamation marks won't be needed.

Reply:
they are still required

Reply:
No if you'll make the options required by removing the question marks from the options typing.

Reply:
If I make them required then what's the point of default values?